Python module needs to be built without these extra renames.
MPI module needs to reference correct python

~Index: boost_1_64_0/libs/mpi/build/Jamfile.v2
~===================================================================
~--- boost_1_64_0.orig/libs/mpi/build/Jamfile.v2
~+++ boost_1_64_0/libs/mpi/build/Jamfile.v2
~@@ -61,10 +61,10 @@ libraries += boost_mpi ;
~     lib boost_mpi_python
~       : # Sources
~         python/serialize.cpp
~+        /user-config//boost_python_alias
~       : # Requirements
~         <library>boost_mpi
~         <library>/mpi//mpi [ mpi.extra-requirements ]
~-      	<library>/boost/python//boost_python
~         <link>shared:<define>BOOST_MPI_DYN_LINK=1
~         <link>shared:<define>BOOST_MPI_PYTHON_DYN_LINK=1
~         <link>shared:<define>BOOST_PYTHON_DYN_LINK=1
~@@ -92,8 +92,8 @@ libraries += boost_mpi ;
~         python/skeleton_and_content.cpp
~         python/status.cpp
~         python/py_timer.cpp
~+        /user-config//boost_python_alias
~       : # Requirements
~-      	<library>/boost/python//boost_python
~         <library>boost_mpi_python
~         <library>boost_mpi
~         <library>/mpi//mpi [ mpi.extra-requirements ]
Index: boost_1_67_0/libs/python/Jamfile
===================================================================
--- boost_1_67_0.orig/libs/python/Jamfile
+++ boost_1_67_0/libs/python/Jamfile
@@ -36,9 +36,10 @@ local rule split-version ( version )
 # For example, Boost.Python built for Python 2.7 uses the suffix "27"
 rule version-suffix ( version )
 {
-    local major-minor = [ split-version $(version) ] ;
-    local suffix = $(major-minor:J="") ;
-    return $(suffix) ;
+    # local major-minor = [ split-version $(version) ] ;
+    # local suffix = $(major-minor:J="") ;
+    # return $(suffix) ;
+    return "" ;
 }


diff -ur boost_1_67_0.orig/boostcpp.jam boost_1_67_0/boostcpp.jam
--- boost_1_67_0.orig/boostcpp.jam	2018-07-30 16:26:03.346187542 -0700
+++ boost_1_67_0/boostcpp.jam	2018-07-30 16:26:18.839081472 -0700
@@ -105,7 +105,7 @@
 python-id = [ option.get "python-buildid" ] ;
 if $(python-id)
 {
-    PYTHON_ID = [ regex.replace $(python-id) "[*\\/:.\"\']" _ ] ;
+    PYTHON_ID = [ regex.replace $(python-id) "[*\\/:\"\']" _ ] ;
 }


diff -ur boost_1_67_0.orig/libs/python/Jamfile boost_1_67_0/libs/python/Jamfile
--- boost_1_67_0.orig/libs/python/Jamfile	2018-07-30 14:56:15.591366992 -0700
+++ boost_1_67_0/libs/python/Jamfile	2018-07-30 16:26:31.662993677 -0700
@@ -47,7 +47,7 @@
 python-id = [ option.get "python-buildid" ] ;
 if $(python-id)
 {
-    PYTHON_ID = [ regex.replace $(python-id) "[*\\/:.\"\']" _ ] ;
+    PYTHON_ID = [ regex.replace $(python-id) "[*\\/:\"\']" _ ] ;
 }

 rule python-tag ( name : type ? : property-set )
--- boost_1_68_0.orig/tools/build/src/tools/python.jam	2018-08-01 21:50:55.000000000 +0100
+++ boost_1_68_0/tools/build/src/tools/python.jam	2018-10-15 09:47:49.946314483 +0100
@@ -661,9 +661,10 @@
 # For example, Boost.Python built for Python 2.7 uses the suffix "27"
 rule version-suffix ( version )
 {
-    local major-minor = [ split-version $(version) ] ;
-    local suffix = $(major-minor:J="") ;
-    return $(suffix) ;
+    #local major-minor = [ split-version $(version) ] ;
+    #local suffix = $(major-minor:J="") ;
+    #return $(suffix) ;
+    return "" ;
 }
 
 # Declare a target to represent Python's library.
